src/Microsoft.Net.Http.Headers/NameValueHeaderValue.cs

```diff
@@ -320,6 +320,25 @@ private static int GetNameValueLength(string input, int startIndex, out NameValu
             }
         }
 
+        internal static NameValueHeaderValue Find(ObjectCollection<NameValueHeaderValue> values, string name)
+        {
+            Contract.Requires((name != null) && (name.Length > 0));
+
+            if ((values == null) || (values.Count == 0))
+            {
+                return null;
+            }
+
+            foreach (var value in values)
+            {
+                if (string.Compare(value.Name, name, StringComparison.OrdinalIgnoreCase) == 0)
```

Comment:
Curious: Is this not `string.Equals()` primarily for consistency with some of the comparisons that pass more arguments to `Compare()`?